src/socket.rs

```diff
@@ -70,6 +89,8 @@ pub struct Socket {
     link: Option<LteLink>,
     /// Gets set to true when the socket has been split. This is relevant for the drop functions
     split: bool,
+    /// When socket is split, this prevents RX/TX wakers from overwriting each other
+    dir: SocketDirection,
```

Comment:
We can save a bit more RAM by not storing this in the socket. (And it might not work with borrowed sockets anyways)

There are two possibilities. Either we have one full socket that can do TX and RX or we have two split sockets one of which can TX and one of which can RX.
This means that there's always at most one socket that can do TX or RX.

With this info, we know that we don't have to keep track of this value, but can pass it along with the operation we do.